src/Integrations/Integrations/WordPress/V6/WordPressComponent.php

```diff
@@ -0,0 +1,480 @@
+            'render_block',
+            function (SpanData $span, $args, $retval) use ($service) {
+                $span->name = 'block';
+                // Some blocks are literally empty. We could even consider dropping the span in this case.
+                $blockName = isset($args[0]['blockName']) ? $args[0]['blockName'] : '?';
+                $span->resource = "(block_name: $blockName)";
+
+                if (isset($args[0]['attrs'])) {
+                    $attrs = $args[0]['attrs'];
+                    foreach (['slug', 'theme', 'area', 'tagName'] as $attr) {
+                        if (isset($attrs[$attr])) {
+                            $span->meta["wp.template_part.$attr"] = $attrs[$attr];
+                        }
+                    }
+                }
+
+                $span->service = $service;
+                $span->meta[Tag::COMPONENT] = WordPressIntegration::NAME;
+            }
+        );
+
+        trace_function('block_template_part', function (SpanData $span, $args) {
+            $span->name = 'block_template_part';
+            $part = isset($args[0]) && is_string($args[0]) ? $args[0] : '?';
+            $span->resource = "(part: $part)";
+        });
+
+        trace_function('get_query_template', function (SpanData $span, $args) use ($service) {
+            $span->type = Type::WEB_SERVLET;
+            $span->name = 'template';
+
+            $type = isset($args[0]) ? $args[0] : '?';
+            $span->resource = "(type: $type)";
+
+            $span->meta['wp.template.type'] = $type; // TODO: Redundant with resource? Search for more relevant tags
+            $span->service = $service;
+            $span->meta[Tag::COMPONENT] = WordPressIntegration::NAME;
+        });
+
+        // Cookies
+        trace_function('setcookie', function (SpanData $span, $args) use ($service) {
+            $span->type = Type::WEB_SERVLET;
+            $span->name = 'setcookie';
+
+            $name = isset($args[0]) ? $args[0] : '?';
+            $span->resource = "(name: $name)";
+
+            $span->meta['wp.cookie.name'] = $name;
+            $span->service = $service;
+            $span->meta[Tag::COMPONENT] = WordPressIntegration::NAME;
+        });
+
+        hook_function('add_action', function ($args) use ($service, &$actionHookToPlugin) {
+            $action = $args[0];
+            $callback = $args[1];
+
+            if (
+                in_array(
+                    $action,
+                    [
+                        'plugins_loaded',
+                        'setup_theme',
+                        'after_setup_theme',
+                        'init',
+                        'wp_loaded',
+                        'template_redirect',
+                        'wp', // part of wp->main();
+                        'wp_head',
+                        'rest_api_init',
+                        'wp_footer',
+                        'shutdown'
+                    ]
+                )
+            ) {
```

Comment:
@bwoebi Do you think this should be converted into a "set"? Something like this:

```suggestion
            $interestingActions = [
                        'plugins_loaded' => true,
                        'setup_theme' => true,
                        'after_setup_theme' => true,
                        'init' => true,
                        'wp_loaded' => true,
                        'template_redirect' => true,
                        'wp' => true, // part of wp->main();
                        'wp_head' => true,
                        'rest_api_init' => true,
                        'wp_footer' => true,
                        'shutdown' => true,
                    ];
            if (isset($interestingActions[$action])) {
```

Reply:
Yes.

Reply:
@PROFeNoM Using `in_array` will be `O(n)` whereas the hashmap masquerading as a set is `O(1)`. Given how frequently this code will be called, that will definitely matter.

Reply:
Yes, this is so much more better. Thanks!

Also, don't you think it would make sense for users to be able to customize this list through a comma-separated list of values through an environment variable ?